src/main/scala/com/microsoft/hyperspace/index/plans/logical/IndexHadoopFsRelation.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (2020) The Hyperspace Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.microsoft.hyperspace.index.plans.logical
+
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.catalog.BucketSpec
+import org.apache.spark.sql.execution.datasources.{FileFormat, FileIndex, HadoopFsRelation}
+import org.apache.spark.sql.types.StructType
+
+import com.microsoft.hyperspace.index.IndexLogEntry
+
+class IndexHadoopFsRelation(
+    spark: SparkSession,
+    index: IndexLogEntry,
+    location: FileIndex,
+    partitionSchema: StructType,
+    dataSchema: StructType,
+    bucketSpec: Option[BucketSpec],
+    fileFormat: FileFormat,
+    options: Map[String, String])
+    extends HadoopFsRelation(
+      location,
+      partitionSchema,
+      dataSchema,
+      bucketSpec,
+      fileFormat,
+      options)(spark) {
+
+  override def toString(): String = {
+    s"Index ${index.name}_${index.logVersion}"
```

Comment:
How about `s"Index(Name: ${index.name}, LogVersion: ${index.logVersion}"` to be more explicit? I am fine with the current approach as well if other people like it.

Reply:
It seems a little long, but I'm also find with either way. @rapoth @pirz @apoorvedave1 WDYT?

current: `Index indexName_1`
suggested: `Index(Name: indexName, LogVersion: 1)`

if index name includes `_`:
current: `Index index_name_1`
suggested: `Index(Name: index_name, LogVersion: 1)`